Tests/Unit/EmogrifierTest.php

```diff
@@ -2104,4 +2101,35 @@ public function emogrifyKeepsInlineStylePriorityVersusStyleBlockRules()
 
         self::assertContains('<p style="padding: 10px; padding-left: 20px;">', $result);
     }
+    
+    /**
+     * @test
+     */
+    public function emogrifyPrefersInlineStyleOverCSSBlockStyleForHtmlAttributesMapping()
+    {
+        $this->subject->setHtml(
+            '<html><head><style>p {width:1px}</style></head><body><p style="width:2px"></p></body></html>'
+        );
+        $this->subject->enableCssToHtmlMapping();
+
+        $result = $this->subject->emogrify();
+
+        self::assertContains('width="2"', $result);
```

Comment:
This should be more specific, i.e., it should test that this is an attribute of the P node.